File: 프로그래머스/lv0/120913. 잘라서 배열로 저장하기/잘라서 배열로 저장하기.java
import java.util.*;
import java.lang.*;
import java.math.*;
import java.io.*;

class Solution {
    public String[] solution(String my_str, int n) {
        String[] answer = {};
        
        int len = 0;
        if(my_str.length() % n == 0) len = my_str.length()/n;
        else len = my_str.length()/n + 1;
        String[] div = new String[len];
        
        for(int i=0; i<div.length; i++){
            String part = "";
            if(my_str.length() >= n){
                part = my_str.substring(0, n);
                my_str = my_str.substring(n, my_str.length());
            }
            else{
                part = my_str;
                my_str = "";
            }
            div[i] = part;
        }
        answer = Arrays.copyOf(div, div.length);
        
        return answer;
    }
}

File: 프로그래머스/lv0/120913. 잘라서 배열로 저장하기/잘라서 배열로 저장하기.cpp
#include <string>
#include <vector>
#include <iostream>

using namespace std;

vector<string> solution(string my_str, int n) {
    vector<string> answer;
    
    while(true){
        
        if(my_str.length() >= n){
            answer.push_back(my_str.substr(0, n));
            my_str = my_str.substr(n);
        }
        else if(my_str.length() == 0){
            break;
        }
        else{
            answer.push_back(my_str);
            break;
        }
        
    }
    
    
    
    
    return answer;
}